include/seqan3/alignment/matrix/detail/affine_cell_proxy.hpp

```diff
@@ -131,16 +132,95 @@ class affine_cell_proxy : public tuple_t
     /*!\name Constructors, destructor and assignment
      * \{
      */
-    affine_cell_proxy()= default; //!< Defaulted.
+    affine_cell_proxy() = default; //!< Defaulted.
     affine_cell_proxy(affine_cell_proxy const &) = default; //!< Defaulted.
     affine_cell_proxy(affine_cell_proxy &&) = default; //!< Defaulted.
     affine_cell_proxy & operator=(affine_cell_proxy const &) = default; //!< Defaulted.
     affine_cell_proxy & operator=(affine_cell_proxy &&) = default; //!< Defaulted.
     ~affine_cell_proxy() = default; //!< Defaulted.
 
-    // Inherit the tuple constructors and assignment.
+    // Inherit the base class's constructor to enable element-wise initialisation (direct and converting constructor).
     using tuple_t::tuple_t;
-    using tuple_t::operator=;
+
+    //!\brief Converting constructor. Initialises from another tuple type.
+    template <typename other_tuple_t>
+    //!\cond
+        requires std::constructible_from<tuple_t, other_tuple_t &&>
+    //!\endcond
+    explicit affine_cell_proxy(other_tuple_t && other) :
+        tuple_t{std::forward<other_tuple_t>(other)}
+    {}
+
+    //!\brief Converting copy-constructor.
+    template <typename other_tuple_t>
+    //!\cond
+        requires std::constructible_from<tuple_t, other_tuple_t &>
+    //!\endcond
+    explicit affine_cell_proxy(affine_cell_proxy<other_tuple_t> & other) :
+        tuple_t{static_cast<other_tuple_t &>(other)}
+    {}
+
+    //!\brief Converting copy-constructor.
+    template <typename other_tuple_t>
+    //!\cond
+        requires std::constructible_from<tuple_t, other_tuple_t const &>
+    //!\endcond
+    explicit affine_cell_proxy(affine_cell_proxy<other_tuple_t> const & other) :
+        tuple_t{static_cast<other_tuple_t const &>(other)}
+    {}
+
+    //!\brief Converting move-constructor.
+    template <typename other_tuple_t>
+    //!\cond
+        requires std::constructible_from<tuple_t, other_tuple_t>
```

Comment:
I know that it is not a forwarding-reference.

I asked because you do

```c++
tuple_t{static_cast<other_tuple_t &&>(some_type)}
```

that means you call the `tuple_t` constructor with the type `other_tuple_t &&`. So you expect that `std::constructible_from<tuple_t, other_tuple_t &&>` is working.

I always forget what prvalues and xvalues are, so not sure what you mean. But anyhow, any sane constructor should work with `const &` and `&&` type modifier.